<?xml version="1.0"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from kuka_model.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="kuka_description" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!--<xacro:include filename="materials.xacro"/>-->
  <gazebo reference="link1">
    <gravity>0</gravity>
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="link2">
    <gravity>0</gravity>
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="link3">
    <gravity>0</gravity>
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="link4">
    <gravity>0</gravity>
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="link5">
    <gravity>0</gravity>
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="link6">
    <gravity>0</gravity>
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="base">
    <material>Gazebo/Black</material>
  </gazebo>
  <gazebo reference="depth_sensor">
    <material>Gazebo/Black</material>
  </gazebo>


  <!-- KUKA limites de software -5 en grados -->
  <!-- Ejes a2 y a3 tienen offset no incluido ver definicion de joints-->
  <!-- ############################################ -->
  <link name="base">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/base_link.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/base_link.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="link1">
    
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_1.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_1.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="a1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.35"/>
    <parent link="base"/>
    <child link="link1"/>
    <axis xyz="0 0 -1"/>
    <!-- Obtenidos de la documentacion KUKA -->
    <limit effort="-1" lower="-3.141594" upper="3.141594" velocity="2.6529"/>
    <dynamics damping="5" friction="5" />
  </joint>
  <link name="link2">
    
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.35"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_2.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_2.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="a2" type="revolute">
    <origin rpy="0 0 0" xyz="0.3 -0.068 0.315"/>
    <parent link="link1"/>
    <child link="link2"/>
    <axis xyz="0 1 0"/>
    <!-- Offset de -90 -->
    <limit effort="-1" lower="-0.698132" upper="1.396264" velocity="2.6529"/>
    <dynamics damping="5" friction="5"/>
  </joint>
  <link name="link3">
    
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_3.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_3.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="a3" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.011 0.65"/>
    <parent link="link2"/>
    <child link="link3"/>
    <axis xyz="0 0.99939 -0.034899"/>
    <!-- Offset de +90 -->
    <limit effort="-1" lower="-3.5779265" upper="1.1344645" velocity="2.6529"/>
    <dynamics damping="5" friction="5" />
  </joint>
  <link name="link4">
    
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_4.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_4.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="a4" type="revolute">
    <origin rpy="0 0 0" xyz="0.358 0.08636 0.15208"/>
    <parent link="link3"/>
    <child link="link4"/>
    <axis xyz="-1 0 0"/>
    <limit effort="-1" lower="-6.0213885" upper="6.0213885" velocity="4.3633"/>
    <dynamics damping="5" friction="5" />
  </joint>
  <link name="link5">
    
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_5.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_5.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="a5" type="revolute">
    <origin rpy="0 0 0" xyz="0.242 0.020238 -0.00070671"/>
    <parent link="link4"/>
    <child link="link5"/>
    <axis xyz="0 1 0"/>
    <limit effort="-1" lower="-1.919863" upper="1.396264" velocity="6.2308"/>
    <dynamics damping="5" friction="5" />
  </joint>
  <link name="link6">
    
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_6.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.74 0.74 0.74 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_6.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="a6" type="revolute">
    <origin rpy="0 0 0" xyz="0.1125 -0.02025 0"/>
    <parent link="link5"/>
    <child link="link6"/>
    <axis xyz="-1 0 0"/>
    <limit effort="-1" lower="-6.0213885" upper="6.0213885" velocity="11.519"/>
    <dynamics damping="5" friction="5" />
  </joint>
  <!-- Include tool -->
  <link name="kinect_base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/tool/visual/kinect_gripper.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.4 0.4 0.4 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/tool/collision/kinect_gripper.stl"/>
      </geometry>
    </collision>
  </link>

  <link name="camera_depth_frame">
        <inertial>
      <mass value="0.001" />
      <origin xyz="0 0 0" />
      <inertia ixx="0.0001" ixy="0.0" ixz="0.0"
               iyy="0.0001" iyz="0.0"
               izz="0.0001" />
    </inertial>
  </link>

  <joint name="kinect_base" type="fixed">
    <origin rpy="0 0 0" xyz="0.095 -0.02025 0"/>
    <parent link="link5"/>
    <child link="kinect_base_link"/>
  </joint>
  
  <joint name="camera_depth_joint" type="fixed">
    <origin rpy="0 1.5435 0" xyz="-0.0013956 0.011693 -0.14997"/>
    <parent link="kinect_base_link"/>
    <child link="camera_depth_frame"/>
  </joint>

  <joint name="camera_depth_optical_joint" type="fixed">
    <origin xyz="0 0 0" rpy="-1.570796 0 -1.570796" />
    <parent link="camera_depth_frame" />
    <child link="camera_depth_optical_frame" />
  </joint>

  <link name="camera_depth_optical_frame">
    <inertial>
      <mass value="0.001" />
      <origin xyz="0 0 0" />
      <inertia ixx="0.0001" ixy="0.0" ixz="0.0"
               iyy="0.0001" iyz="0.0"
               izz="0.0001" />
    </inertial>
  </link>


  <gazebo reference="camera_depth_frame">  
      <sensor type="depth" name="camera">
        <always_on>true</always_on>
        <update_rate>5</update_rate>
        <camera>
          <horizontal_fov>1.04719755</horizontal_fov>
          <image>
            <format>R8G8B8</format>
            <width>640</width>
            <height>480</height>
          </image>
          <clip>
            <near>0.05</near>
            <far>8.0</far>
          </clip>
        </camera>
        <plugin name="kinect_camera_controller" filename="libgazebo_ros_openni_kinect.so">
          <cameraName>camera</cameraName>
          <alwaysOn>true</alwaysOn>
          <updateRate>2</updateRate>
          <imageTopicName>rgb/image_raw</imageTopicName>
          <depthImageTopicName>depth/image_raw</depthImageTopicName>
          <pointCloudTopicName>depth/points</pointCloudTopicName>
          <cameraInfoTopicName>rgb/camera_info</cameraInfoTopicName>
          <depthImageCameraInfoTopicName>depth/camera_info</depthImageCameraInfoTopicName>
          <frameName>real/camera_depth_optical_frame</frameName>
          <baseline>0.1</baseline>
          <distortion_k1>0.0</distortion_k1>
          <distortion_k2>0.0</distortion_k2>
          <distortion_k3>0.0</distortion_k3>
          <distortion_t1>0.0</distortion_t1>
          <distortion_t2>0.0</distortion_t2>
          <pointCloudCutoff>0.4</pointCloudCutoff>
        </plugin>
      </sensor>
    </gazebo>

  <gazebo>
    <plugin filename="libkuka_gazebo_joint_controller.so" name="kuka_gazebo_joint_controller">
      <topicName>joint_trajectory</topicName>
      <updateRate>30.0</updateRate>
      <robotNamespace>/kuka_gazebo</robotNamespace>
    </plugin>
  </gazebo>

  <link name="world"/>

  <joint name="fixed" type="fixed">
    <parent link="world"/>
    <child link="base"/>
  </joint>
</robot>